workflows/smrnaseq.nf

```diff
@@ -0,0 +1,241 @@
+
+    reads_for_mirna = FASTQC_TRIMGALORE.out.reads
+    MIRNA_QUANT (
+        reference_mature,
+        reference_hairpin,
+        mirna_gtf,
+        reads_for_mirna
+    )
+    ch_software_versions = ch_software_versions.mix(MIRNA_QUANT.out.bowtie_versions.ifEmpty(null))
+    ch_software_versions = ch_software_versions.mix(MIRNA_QUANT.out.samtools_versions.first().ifEmpty(null))
+    ch_software_versions = ch_software_versions.mix(MIRNA_QUANT.out.seqcluster_versions.first().ifEmpty(null))
+    ch_software_versions = ch_software_versions.mix(MIRNA_QUANT.out.mirtop_versions.ifEmpty(null))
+    ch_software_versions = ch_software_versions.mix(MIRNA_QUANT.out.merge_versions.ifEmpty(null))
+
+    //
+    // GENOME
+    //
+    genome_stats = Channel.empty()
+    if (fasta){
+        fasta_ch = file(fasta)
+        GENOME_QUANT ( fasta_ch, bt_index, MIRNA_QUANT.out.unmapped )
+        GENOME_QUANT.out.stats
+            .set { genome_stats }
+        MIRDEEP2 (FASTQC_TRIMGALORE.out.reads, GENOME_QUANT.out.fasta , GENOME_QUANT.out.indices, MIRNA_QUANT.out.fasta_hairpin, MIRNA_QUANT.out.fasta_mature)
+        ch_software_versions = ch_software_versions.mix(MIRDEEP2.out.versions.first().ifEmpty(null))
+
+    }
+
+    //
+    // MODULE: Pipeline reporting
+    //
+    ch_software_versions
+        .map { it -> if (it) [ it.baseName, it ] }
+        .groupTuple()
+        .map { it[1][0] }
+        .flatten()
+        .collect()
+        .set { ch_software_versions }
+
+    GET_SOFTWARE_VERSIONS (
+        ch_software_versions.map { it }.collect()
+    )
+
+    //
+    // MODULE: MultiQC
+    //
+    workflow_summary    = WorkflowSmrnaseq.paramsSummaryMultiqc(workflow, summary_params)
+    ch_workflow_summary = Channel.value(workflow_summary)
+
+    ch_multiqc_files = Channel.empty()
+    ch_multiqc_files = ch_multiqc_files.mix(Channel.from(ch_multiqc_config))
+    ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_custom_config.collect().ifEmpty([]))
+    ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))
+    ch_multiqc_files = ch_multiqc_files.mix(GET_SOFTWARE_VERSIONS.out.yaml.collect())
+    ch_multiqc_files = ch_multiqc_files.mix(FASTQC_TRIMGALORE.out.fastqc_zip.collect{it[1]}.ifEmpty([]))
+    ch_multiqc_files = ch_multiqc_files.mix(MIRNA_QUANT.out.mature_stats.collect({it[1]}).ifEmpty([]))
+    ch_multiqc_files = ch_multiqc_files.mix(MIRNA_QUANT.out.hairpin_stats.collect({it[1]}).ifEmpty([]))
+    ch_multiqc_files = ch_multiqc_files.mix(genome_stats.collect({it[1]}).ifEmpty([]))
+    ch_multiqc_files = ch_multiqc_files.mix(MIRNA_QUANT.out.mirtop_logs.collect().ifEmpty([]))
+    ch_multiqc_files = ch_multiqc_files.mix(MIRTRACE.out.results.collect().ifEmpty([]))
+
+    MULTIQC (
+        ch_multiqc_files.collect()
+    )
+    multiqc_report       = MULTIQC.out.report.toList()
+    ch_software_versions = ch_software_versions.mix(MULTIQC.out.versions.ifEmpty(null))
```

Comment:
You may need to add `CUSTOM_DUMPSOFTWAREVERSIONS` output as input, either as a channel such as in `rnaseq` [pipeline](https://github.com/nf-core/rnaseq/blob/964425e3fd8bfc3dc7bce43279a98d17a874d3f7/workflows/rnaseq.nf#L769) or by mixing it with `ch_mulitqc_files` as is done here
```suggestion
```